### PR TITLE
Add status check before deserializing runner-registration response

### PR DESCRIPTION
**Pull request by nikola-jokic**
The purpose of this PR is to ensure the listener does not panic if the status code returned by the actions service is not in 2xx range.

Previously, we would try to unmarshal JSON which would leave nil values for actions service admin token. When later used, the listener would panic.

Added check similar to one in the runner making sure the status code is success, and log the error message if returned from the actions service

https://github.com/actions/runner/blob/9c57205c1c1cd7ada73cb08b556293ec0411f8d4/src/Runner.Listener/Configuration/ConfigurationManager.cs#L777C50-L777C50

### CODE REVIEW

File: github/actions/testserver/server.go [hunk 1]
```diff
@@ -73,10 +70,45 @@ func WithActionsToken(token string) actionsServerOption {
...
+
+	if s.actionRegistrationTokenHandler == nil {
+		s.actionRegistrationTokenHandler = func(w http.ResponseWriter, r *http.Request) {
+			if s.token == "" {
+				s.token = DefaultActionsToken(t)
+			}
+
+			w.WriteHeader(http.StatusCreated)
+			w.Write([]byte(`{"url":"` + s.URL + `/tenant/123/","token":"` + s.token + `"}`))
+			return
```
Comment (nikola-jokic) [inline github/actions/testserver/server.go]:
```suggestion

```

File: github/actions/client.go [hunk 2]
```diff
@@ -921,6 +921,10 @@ func (c *Client) getActionsServiceAdminConnection(ctx context.Context, rt *regis
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return nil, fmt.Errorf("unexpected response from Actions service during registration call: %v", resp.StatusCode)
```
Comment (Link-) [inline github/actions/client.go]: Are we dumping that response body somewhere else? We might need to see what the service is returning in order to troubleshoot the root cause of this error

**Review by Link- — CHANGES_REQUESTED** [review]
Can we please add more context to the PR description so that we use it as documentation for why this change was introduced? 

1. Describe the problem
2. How this change addresses that problem

**Review by Link- — APPROVED** [review]
This looks good, thank you

**Pull request merged**